include/clspv/Option.h

```diff
@@ -16,10 +16,32 @@
 #define CLSPV_INCLUDE_CLSPV_OPTION_H_
 
 #include <cstdint>
+#include <set>
+
+#include "llvm/Support/CommandLine.h"
 
 namespace clspv {
+
+namespace Builtins {
+enum BuiltinType : unsigned int;
+}
+
 namespace Option {
 
+// Custom parser for parsing a comma separated list of builtin functions into a
+// std::set containing the equivalent list of BuiltinType enums. We use a
+// custom parser with a regular cl::opt instead of a cl::list because cl::list
+// requires you to list all possible choices (in this case all builtin
+// functions) and then includes all those choices in the helptext for the flag.
+struct BuiltinListParser
```

Comment:
This seems unnecessary in the header. Can remove the command line header include too.